cypress/integration/projectSettings/project_settings.ts

```diff
@@ -376,9 +376,22 @@ describe("Project Settings when not defaulting to repo", () => {
     it("Should show three populated fields when navigating back from another page", () => {
       cy.dataCy("navitem-access").click();
       cy.dataCy("navitem-variables").click();
-      cy.dataCy("var-name-input").eq(0).should("have.value", "admin_var");
-      cy.dataCy("var-name-input").eq(1).should("have.value", "sample_name");
-      cy.dataCy("var-name-input").eq(2).should("have.value", "sample_name_2");
+      cy.dataCy("var-name-input").should("be.visible");
+      cy.dataCy("var-name-input").should(($inputs) => {
+        expect($inputs).to.have.length(3);
+        // Asserts input copy exists without asserting input order
+        // to avoid test flake if the input order rearranges.
+        const inputVals = new Set([
+          $inputs.eq(0).val(),
+          $inputs.eq(1).val(),
+          $inputs.eq(2).val(),
+        ]);
+        const copy = new Set(["admin_var", "sample_name", "sample_name_2"]);
+        expect(inputVals.size).to.equal(copy.size);
+        expect(
+          Array.from(inputVals).every((value) => copy.has(value))
+        ).to.equal(true);
+      });
```

Comment:
[`.each()`](https://docs.cypress.io/api/commands/each#Syntax) seems suited for this use case. 
I am confused as to why these tests would start to flake however?

Reply:
We need to wrap the input values in a set to make sure they are each accounted for. We could use `.each` but we would need to create an outside set or counter to make sure all 3 values exist in the tested inputs. I think it's better to use `.should` and have the set logic contained in there.